apps/pokedex/pokedex.star

```diff
@@ -52,3 +55,34 @@ def main():
 def random():
     """Return a pseudo-random number in [0, 1)"""
     return time.now().nanosecond / (1000 * 1000 * 1000)
+
+def get_pokemon(id):
+    key = "pokemon_{}".format(id)
+    url = POKEAPI_URL.format(id)
+
+    data = cache.get(key)
+    if data != None:
+        return json.decode(data)
+
+    res = http.get(url = url)
+    if res.status_code != 200:
+        fail("request to %s failed with status code: %d - %s" % (url, res.status_code, res.body()))
+
+    cache.set(key, json.encode(res.json()), ttl_seconds = CACHE_TTL_SECONDS)
+
+    return res.json()
+
+def get_sprite(url):
+    key = "sprite_{}".format(base64.encode(url))
+
+    data = cache.get(key)
+    if data != None:
+        return base64.decode(data)
+
+    res = http.get(url = url)
+    if res.status_code != 200:
+        fail("request to %s failed with status code: %d - %s" % (url, res.status_code, res.body()))
+
+    cache.set(key, base64.encode(res.body()), ttl_seconds = CACHE_TTL_SECONDS)
+
+    return res.body()
```

Comment:
This method looks very similar to `get_pokemon`, sans a few minor changes. It'd be nice if this code existing as a library so each app didn't need to implement similar logic, especially logic that's somewhat error prone:
- Need to use the same `key` for `get` and `set`
- Need to return the same value passed to `set`
- Need to be mindful of the encoding scheme

Reply:
Totally. We need to support a starlark library in pixlet so we can have a common set of modules. a `get_cachable_url` method definitely feels useful. Something like:
```
get_cachable_data(url, ttl_seconds)
```

I can follow up on this PR to at least provide something that can be copy/pastad and dry out this code.